src/dbnode/storage/block/types.go

```diff
@@ -168,10 +168,6 @@ type DatabaseBlock interface {
 	// merged during Stream().
 	HasMergeTarget() bool
 
-	// IsRetrieved returns whether the block is already retrieved. Only
```

Comment:
Hm, shouldn't this be required for some cache policies (e.g. LRU/RecentlyRead)?

Reply:
I trusted the comment! I believe the other caching policies use `IsCachedBlock()` and "retrieved" means we have metadata but we may not have the data for it.

Reply:
Yeah I think `IsRetrieved` was specific to that policy (you can see the same logic implemented in the `IsCachedBlock` function. I think we definitely want to get rid of this function and move towards using `IsCachedBlock` or honestly at this point we can delete `IsCachedBlock` as well and only use `WasRetrievedFromDisk since they will have the exact same logic at that point. The only place we use `IsCachedBlock` is in `series.FetchBlocksMetadata` and its just an optimization there (and not a particularly accurate one at that since it doesn't work in the `recently_read` policy)

@robskillington

Reply:
Yeah let's remove it, don't think this is plausible now.